packages/client/src/clients/guide/client.ts

```diff
@@ -76,18 +82,60 @@ const checkForWindow = () => {
 export const guidesApiRootPath = (userId: string | undefined | null) =>
   `/v1/users/${userId}/guides`;
 
-// Detect debug params like "knock_guide_key" from URL.
+// Detect debug params from URL or local storage
 const detectDebugParams = (): DebugState => {
   const win = checkForWindow();
   if (!win) {
     return { forcedGuideKey: null, previewSessionId: null };
   }
 
   const urlParams = new URLSearchParams(win.location.search);
-  const forcedGuideKey = urlParams.get(DEBUG_QUERY_PARAMS.GUIDE_KEY);
-  const previewSessionId = urlParams.get(DEBUG_QUERY_PARAMS.PREVIEW_SESSION_ID);
+  const urlGuideKey = urlParams.get(DEBUG_QUERY_PARAMS.GUIDE_KEY);
+  const urlPreviewSessionId = urlParams.get(
+    DEBUG_QUERY_PARAMS.PREVIEW_SESSION_ID,
+  );
+
+  // If URL params exist, persist them to localStorage and return them
+  if (urlGuideKey || urlPreviewSessionId) {
+    if (win.localStorage) {
+      try {
+        win.localStorage.setItem(
+          DEBUG_STORAGE_KEYS.GUIDE_KEY,
+          urlGuideKey || "",
+        );
+        win.localStorage.setItem(
+          DEBUG_STORAGE_KEYS.PREVIEW_SESSION_ID,
+          urlPreviewSessionId || "",
+        );
+      } catch {
+        // Silently fail in privacy mode
```

Comment:
Seems rare, but local storage can throw in some browsers or if storage is full. 

Also, I opted to just repeat these trycatch & window checks since we only have a few local storage operations, but we could refactor these into a shared util to clean it up.